Makefile

```diff
@@ -235,9 +235,26 @@ install-dunstify: dunstify
 	install -Dm755 dunstify ${DESTDIR}${BINDIR}/dunstify
 endif
 
+ifneq (0,${COMPLETIONS})
+install: install-completions
+install-completions:
+	install -Dm644 completions/dunst.bashcomp ${DESTDIR}${BASHCOMPLETIONDIR}/dunst
+	install -Dm644 completions/dunstctl.bashcomp ${DESTDIR}${BASHCOMPLETIONDIR}/dunstctl
+	install -Dm644 completions/_dunst.zshcomp ${DESTDIR}${ZSHCOMPLETIONDIR}/_dunst
+	install -Dm644 completions/_dunstctl.zshcomp ${DESTDIR}${ZSHCOMPLETIONDIR}/_dunstctl
+	install -Dm644 completions/dunst.fishcomp ${DESTDIR}${FISHCOMPLETIONDIR}/dunst
+	install -Dm644 completions/dunstctl.fishcomp ${DESTDIR}${FISHCOMPLETIONDIR}/dunstctl
```

Comment:
stupid question: what happens if I don't have fish shell? does this command fail or does it do nothing?

Reply:
It just copies a file during `make install`. None of the shells is actually involved at this stage.

Reply:
So if the path doesn't exist it does nothing 👍

Reply:
I'm not sure what you mean by that, but the path will be created within `$DESTDIR` and the file will be put there. It's just that the code therein isn't run.

Reply:
Ah, I was just wondering what happened if the path was not found. So it creates the dirs

Reply:
What about doing a simple check like

```make
install-completions:
	install -Dm644 completions/dunst.bashcomp ${DESTDIR}${BASHCOMPLETIONDIR}/dunst
	install -Dm644 completions/dunstctl.bashcomp ${DESTDIR}${BASHCOMPLETIONDIR}/dunstctl
	[ -d "${DESTDIR}${ZSHCOMPLETIONDIR}" ] && install -Dm644 completions/_dunst.zshcomp ${DESTDIR}${ZSHCOMPLETIONDIR}/_dunst
	[ -d "${DESTDIR}${FISHCOMPLETIONDIR}" ] && install -Dm644 completions/_dunstctl.zshcomp ${DESTDIR}${ZSHCOMPLETIONDIR}/_dunstctl
	[ -d "${DESTDIR}${FISHCOMPLETIONDIR}" ] && install -Dm644 completions/dunst.fishcomp ${DESTDIR}${FISHCOMPLETIONDIR}/dunst
	[ -d "${DESTDIR}${FISHCOMPLETIONDIR}" ] && install -Dm644 completions/dunstctl.fishcomp ${DESTDIR}${FISHCOMPLETIONDIR}/dunstctl

```

Reply:
I don't see why this check should be needed. The `-D` flag of `install` already takes care to make sure that the path exists before copying over the file.

Reply:
I think, we have a misunderstanding what 

> make sure that the path exists

means. I use it in the way, that the path is created (not checked for its existence) before any files are copied and this is also what is documented in the man page of `install`.

If you goal is to skip installation of bash, fish or zsh completion if any shell is not installed, this won't generally work as `$DESTDIR` is a common idiom used e.g. in packaging where an empty directory is used (to avoid artifacts in the package). Thus existence checks will never succeed and thus no shell completions would ever be installed.

Reply:
Right, my fault 🤣
Go ahead and merge

Reply:
You should be able to do the merging @bynect. You can do that when you think it's ready